### ISSUE TITLE
Allow for Setting the Priority as "explicit"

### ISSUE THREAD

**Issue by dtoniolo** [celsiusnarhwal/poetry-source-env#2]
Hi, thanks for this awesome plugin. I needed to set an explicit index for some of my packages and I couldn't do it with Poetry, PDM or any other tool, being able to add your plugin to Poetry has saved my day.

I'm opening this issue to warn you and everybody who's interested that currently this plugin does not work with sources that have their priority set to `"explicit"`, e.g.:

```toml
[[tool.poetry.source]]
name = "${FOO_INDEX_NAME}"
url = "${FOO_INDEX_URL}"
priority = "explicit"
```

I believe this is happening because [`poetry.repositories`](https://github.com/celsiusnarhwal/poetry-source-env/blob/a17abab797302d10993f461fa497a1680ef4273f/poetry_source_env/plugin.py#L63) does not return explicit sources, but I'm not sure, because I'm not familiar with the internals of Poetry.

I might give it a try this weekend, thanks again

**Comment by celsiusnarhwal** [comment]
Hi, thanks for opening this. You're correct; [`RepositoryPool.repositories` is hardcoded to not return explicit sources](https://github.com/python-poetry/poetry/blob/6d57e843b4a709f5a5156c9cc5661cdd0f718979/src/poetry/repositories/repository_pool.py#L76).

This issue is fixed in poetry-source-env 2.0.1. Update the plugin and let me know if thing still aren't working.

**Comment by dtoniolo** [comment]
Just tested it and it works perfectly, thanks for your work and your fast answer!
